crates/ruma-common/src/identifiers/macros.rs

```diff
@@ -24,8 +24,87 @@ macro_rules! partial_eq_string {
     }
 }
 
+macro_rules! owned_identifier {
+    ($owned:ident, $id:ident) => {
+        #[doc = concat!("Owned variant of ", stringify!($id))]
```

Comment:
Can we add some docs describing how to configure the "storage type" to be used? (i.e. `RUSTFLAGS` / `.cargo/config.toml`)

```rust
        #[doc = concat!("Owned variant of ", stringify!($id))]
        ///
        /// <docs here>
```